projects/georeview/index.js

```diff
@@ -0,0 +1,250 @@
+        div.innerHTML = createReview(item);
+        reviewList.appendChild(div);
+      }
+    }
+
+    return container.innerHTML;
+  };
+
+  const createReview = (data) => {
+    const { name, place, text, time } = data.review;
+
+    const review = `
+      <div class="commentItem">
+        <div class="commentItem__head">
+          <span class="commentItem__autor">${name}</span>
+          <span class="commentItem__info">${place}</span>
+          <span class="commentItem__info">${time}</span>
+        </div>
+        <div class="commentItem__text">${text}</div>
+      </div>
+      `;
+    return review;
+  };
+
+  const addReview = async (e) => {
+    e.preventDefault();
+    if (e.target.dataset.action === 'show-reviews') {
+      const button = e.target;
+      const coords = JSON.parse(button.dataset.coords);
+      const reviews = storageGetData(coords);
+      const form = createForm(coords, reviews);
+      openBalloon(coords, form);
+    }
+    if (e.target.dataset.action === 'add-review') {
+      // ищем форму
+      const form = document.querySelector('form[data-form=add-review]');
+      // получаем её координаты
+      const coords = JSON.parse(form.dataset.coords);
+
+      const address = await getAddress(coords);
+      // получаем остальные данные отзыва
+      const data = {
+        coords: coords,
+        address: address,
+        review: {
+          name: form.elements.name.value,
+          place: form.elements.place.value,
+          text: form.elements.review.value,
+          time: getDateNow(),
+        },
+      };
+      storageAddReview(data);
+      createPlacemark(coords);
+      closeBalloon();
+    }
+  };
+
+  document.addEventListener('click', addReview);
+
+  // получить отзывы по координатам из localStorage
+  const storageGetData = (coords = '') => {
+    let data = JSON.parse(localStorage.getItem(storageName));
+
+    if (coords.length > 0) {
+      data = data.filter((element) => element.coords.join('') === coords.join(''));
+    }
+
+    return data;
+  };
+
+  // добавить ревью в localStorage
+  const storageAddReview = async (data) => {
+    let reviews = storageGetData();
+    if (reviews === null) {
+      reviews = [];
+    }
+    reviews.push(data);
+    localStorage.setItem(storageName, JSON.stringify(reviews));
+  };
+
+  const onInit = () => {
+    const coords = storageGetData();
+    if (coords === null) {
+      return false;
+    }
+    for (const item of coords) {
+      createPlacemark(item.coords);
+    }
+  };
+  onInit();
+}
+
+function getDateNow() {
+  const date = new Date();
+  const day = date.getDate() < 10 ? `0${date.getDate()}` : date.getDate();
+  const month =
+    date.getMonth() + 1 < 10 ? `0${date.getMonth() + 1}` : date.getMonth() + 1;
+
+  return `${day}.${month}.${date.getFullYear()}`;
+}
+
+async function getAddress(coords) {
+  const address = await ymaps.geocode(coords);
+  const res = address.geoObjects.get(0).getAddressLine();
+
+  return res;
```

Comment:
Тут тоже можно сократить: 
```suggestion
  return  address.geoObjects.get(0).getAddressLine();
```